ClosedXML_Tests/Excel/ConditionalFormats/ConditionalFormatTests.cs

```diff
@@ -21,5 +21,41 @@ public void MaintainConditionalFormattingOrder()
                 }, @"Other\StyleReferenceFiles\ConditionalFormattingOrder\ConditionalFormattingOrder.xlsx");
             }
         }
+
+        [Test]
+        public void OptionalRangeConsolidationWithoutConsolidation()
+        {
+            var options = new SaveOptions
+            {
+                ConsolidateConditionalFormatRanges = false,
+                ConsolidateDataValidationRanges = false
+            };
+
+            using (var stream = TestHelper.GetStreamFromResource(TestHelper.GetResourcePath(@"Other\StyleReferenceFiles\ConditionalFormattingOptional\inputfile_multiple_ranges.xlsx")))
+            using (var ms = new MemoryStream())
+            {
+                TestHelper.CreateAndCompare(() =>
```

Comment:
These CreateAndCompare tests bring a lot of headaches because every time the saving logic changes many of such tests start to fail even if the changes have no relation to a feature under testing.

This is why I recommend avoiding such kinds of tests if possible and leave them for complicated cases only.

So, if you are intended to check that `ConsolidateConditionalFormatRanges = false` turns the consolidation off, then just save the file to a memory stream, load it from there and ensure a number of conditional formats is 2. Same for data validations. This way the test will be much easier to support.

Reply:
test implementation updated